Neos.Neos/Classes/Fusion/Helper/NodeHelper.php

```diff
@@ -108,6 +112,14 @@ public function labelForNode(Node $node): NodeLabelToken
         return new NodeLabelToken($node);
     }
 
+    /**
+     * Renders the actual node label based on the NodeType definition in Fusion.
+     */
+    public function getLabel(Node $node): string
```

Comment:
it might be unlucky that the `Neos.Node` is also used as helper to generate the actual label in yaml. See `NodeHelper::labelForNode` above. Like:

```
'Vendor.Site:MyContent':
  label: "${Neos.Node.labelForNode(node).prefix('foo')}"
```

so its unfortunate that the naming is not distinct and one could easily create a recursion here. Though that was also possible in 8.3 by just writing `label: "${node.label}"`

As an outlook i would propose to deprecate `Neos.Node` for the yaml case at some point and then introduce a more specialised `Label.createForNode` helper or something that is only to be used in YAML. Having the capabilities to access the "NodeLabelToken" through fusion is not the desired usecase as far as i can tell.

Thats why i named the method `getLabel` in the first place and not just `label` as the ambiguity would be even bigger. Though `Neos.Node.label(node)` would fit better with the semantics `Neos.Node.depth(node)` and `path` so we have to decide on that!

Reply:
So you want to unify them into one?

Reply:
I want to split them as they are separate usecases: https://github.com/neos/neos-development-collection/pull/5277/commits/f22d7401dc561a7044ce86c23bb508ebe74637d9

The labelForNode feature was 'just' introduced in https://github.com/neos/neos-development-collection/pull/3328 and i consider it safe that it was not really used in fusion so i have no b/c layer in place.

Reply:
well as its `Neos.Neos.label` i might have messed up the `Neos.Node.getNodeType` earlier so i created a sister pr https://github.com/neos/neos-development-collection/pull/5287